src/app/containers/ChartbeatAnalytics/canonical/index.test.jsx

```diff
@@ -55,9 +57,64 @@ describe('CanonicalChartbeatAnalytics', () => {
       sections: 'section1 section2',
       chartbeatUID: 1111,
       useCanonical: true,
-      virtualReferrer: '/some-path',
-      title: 'This is another article',
+      virtualReferrer: '/page-A',
+      title: 'Page B',
       uid: 123,
     });
   });
+
+  it('should report correct virtual referrer when user navigates back from onward journey', () => {
+    // initial PWA load (Page A)
+    const wrapper = mount(
+      <CanonicalChartbeatAnalytics
+        chartbeatConfig={config}
+        chartbeatSource="//chartbeat.js"
+      />,
+    );
+
+    const pageBConfig = {
```

Comment:
We can refactor the page configs to use a template or function since most parameters are the same.